### PR TITLE
FieldFilter ignores record if no committedData

### PR DESCRIPTION
**Pull request by cnrudd**
Resolves https://github.com/xh/hoist-react/issues/2925

Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [x] Caught up with `develop` branch as of last change.
- [x] Added CHANGELOG entry, or determined not required.
- [x] Reviewed for breaking changes: no breaking changes.
- [x] Updated doc comments: not required.
- [x] Reviewed and tested on Mobile: not required.
- [x] Created Toolbox branch: Can be tested on these 2 toolbox branches:
https://github.com/xh/toolbox/tree/filterCommitted  (Filter Chooser)
https://github.com/xh/toolbox/tree/columnFiltersOnInlineEditingPanel  (Column Filters)

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: data/filter/FieldFilter.js [hunk 1]
```diff
@@ -96,48 +96,66 @@ export class FieldFilter extends Filter {
         switch (op) {
             case '=':
                 return r => {
+                    if (isNil(r.committedData)) return true; // Ignore (do not filter out) record if not committed
```
Comment (TomTirapani) [inline data/filter/FieldFilter.js:99]: I think we need to change these conditions to `if (store && isNil(r.committedData))` - filters are supposed to be able to work against anonymous objects as well as StoreRecords (see [getTestFn docs](https://github.com/xh/hoist-react/blob/develop/data/filter/Filter.js#L28-L30)). As part of that change, we may as well put the condition into a named function e.g. `isUncommittedRecord(r)`.

File: data/filter/FieldFilter.js [hunk 2]
```diff
@@ -80,7 +80,7 @@ export class FieldFilter extends Filter {
 
         if (store) {
             const storeField = store.getField(field);
-            if (!storeField) return () => true; // Ignore if field not in store
+            if (!storeField) return () => true; // Ignore (do not filter out) if field not in store
```
Comment (TomTirapani) [inline data/filter/FieldFilter.js:83]: Great to clarify what 'ignore' means in this context 👍

**Review by TomTirapani — CHANGES_REQUESTED** [review]

File: data/filter/FieldFilter.js (hunk 1 above)
Comment (cnrudd) [reply data/filter/FieldFilter.js]: re `store`: yes, of course, thanks for pointing that out.  I have made the fix and moved it all into a new local function called `mustIgnore`.

**Comment by lbwexler** [comment]
Looks great guys, thanks!

**Pull request merged**